pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -201,16 +202,50 @@ func (is *InstanceService) InstanceCreate(name string, config *openstackconfigv1
 		}
 	}
 	userData := base64.StdEncoding.EncodeToString([]byte(cmd))
+	// if we are using Boot-from-volume, let's ignore image in base create option
+	image := config.Image
+	if config.RootVolume.Size > 0 {
+		image = ""
+	}
+
 	createOpts = servers.CreateOpts{
 		Name:             name,
-		ImageName:        config.Image,
+		ImageName:        image,
 		FlavorName:       config.Flavor,
 		AvailabilityZone: config.AvailabilityZone,
 		Networks:         nets,
 		UserData:         []byte(userData),
 		SecurityGroups:   config.SecurityGroups,
 		ServiceClient:    is.computeClient,
 	}
+
+	if config.RootVolume.Size > 0 {
+		var blocks []bootfromvolume.BlockDevice
+		block := bootfromvolume.BlockDevice{
+			SourceType:          bootfromvolume.SourceImage,
+			BootIndex:           0,
+			UUID:                config.Image,
+			DeleteOnTermination: true,
+			DestinationType:     bootfromvolume.DestinationVolume,
+			VolumeSize:          config.RootVolume.Size,
+			DeviceType:          config.RootVolume.VolumeType,
+		}
+		blocks = append(blocks, block)
+
+		createOptswithbfv := bootfromvolume.CreateOptsExt{
```

Comment:
How about using `createOpts` as `var createOpts servers.CreateOptsBuilder`. This would save us the second place to call `servers.Create`
```
func (is *InstanceService) InstanceCreate(name string, [...]) {
        var createOpts servers.CreateOptsBuilder
        [...]


    	createOpts = servers.CreateOpts{
		Name:             name,
		ImageName:        image,
		FlavorName:       config.Flavor,
		AvailabilityZone: config.AvailabilityZone,
		Networks:         nets,
		UserData:         []byte(userData),
		SecurityGroups:   config.SecurityGroups,
		ServiceClient:    is.computeClient,
	}

        if config.RootVolume.Size > 0 {
                [...]

                 createOpts = bootfromvolume.CreateOptsExt{
			CreateOptsBuilder: createOpts,
			BlockDevice:       blocks,
		}
	}

	server, err := servers.Create(is.computeClient, keypairs.CreateOptsExt{
		CreateOptsBuilder: createOpts,
		KeyName:           keyName,
	}).Extract()
	if err != nil {
		return nil, fmt.Errorf("Create new server err: %v", err)
	}
	return serverToInstance(server), nil
}
```

Reply:
ok, will do